internal/tiltfile/k8s.go

```diff
@@ -692,6 +783,50 @@ func (s *tiltfileState) k8sResourceAssemblyVersionFn(thread *starlark.Thread, fn
 	return starlark.None, nil
 }
 
+func (s *tiltfileState) calculateResourceNames(workloads []k8s.K8sEntity) ([]string, error) {
+	if s.workloadToResourceFunction.fn != nil {
+		names, err := s.workloadToResourceFunctionNames(workloads)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error applying workload_to_resource_function %s", s.workloadToResourceFunction.pos.String())
+		}
+		return names, nil
+	} else {
+		return uniqueResourceNames(workloads)
+	}
+}
+
+// calculates names for workloads using s.workloadToResourceFunction
+func (s *tiltfileState) workloadToResourceFunctionNames(workloads []k8s.K8sEntity) ([]string, error) {
+	takenNames := make(map[string]k8s.K8sEntity)
+	ret := make([]string, len(workloads))
+	thread := s.starlarkThread()
+	for i, e := range workloads {
+		id := newK8SObjectID(e)
+		name, err := s.workloadToResourceFunction.fn(thread, id)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error determing resource name for '%s'", id.String())
+		}
+
+		if conflictingWorkload, ok := takenNames[name]; ok {
+			return nil, fmt.Errorf("both '%s' and '%s' mapped to resource name '%s'", newK8SObjectID(e).String(), newK8SObjectID(conflictingWorkload).String(), name)
```

Comment:
can you add a test that triggers this error?

Reply:
that's [here](https://github.com/windmilleng/tilt/blob/95118527ad3f79dc992212664cbdac3842719039/internal/tiltfile/tiltfile_test.go#L2781)